File: FlowerShop/src/app/services/order.ts
import { ItemOrderDto } from 'src/app/services/cart';

export interface ProductOrderDto {
  deliveryAddress: string,
  deliveryName: string,
  email: string,
  id?: number,
  orderStatus?: OrderStatus,
  paymentType: PaymentType,
  phone: string,
  productItems: ItemOrderDto[],
  text?: string
}

export enum OrderStatus {
  'DELIVERY', 'ENTERING', 'PENDING_PAYMENT', 'PROCESSED', 'PROCESSING'
}

export enum PaymentType {
  CARD = 'CARD', CASH = 'CASH', ONLINE = 'ONLINE'
}

File: FlowerShop/models/dto/Order.cs
namespace FlowerShop.models.dto
{
    public class Order
    {
        public string deliveryAddress { get; set; }
        public string deliveryName { get; set; }
        public string email { get; set; }
        public int id { get; set; }
        public string orderStatus { get; set; }
        public string paymentType { get; set; }
        public string phone { get; set; } 
        public ItemOrder[] productItems { get; set; }
        public string text { get; set; }
    }
}
